libs/sdk-backend-tiger/src/convertors/toBackend/afm/AfmFiltersConverter.ts

```diff
@@ -0,0 +1,105 @@
+        ? [measure, undefined]
+        : [localIdRef(measureLocalId(updatedFilteredMeasure)), false];
+}
+
+/**
+ * Find those ratio measures which are used in some measure based filter. For each such measure create its updated
+ * version with ratio computation off and with new local identifier. These new auxiliary measures are meant to be used
+ * as actual base for their respective filters which should be applied on source data (instead of the result) so that
+ * both numerator and denominator of the ratio measure are actually filtered.
+ *
+ * Return mapping from the original local identifiers to new auxiliary measures.
+ */
+function determineComputeRatioMeasureNumerators(
+    afmMeasures: IMeasure[],
+    filters: IFilter[],
+): ComputeRatioMeasureNumerators {
+    const filteredLocalMeasures = filters
+        .map(filterMeasureRef)
+        .filter(isLocalIdRef)
+        .map((m) => m.localIdentifier);
+    return new Map(
+        afmMeasures
+            .filter(measureDoesComputeRatio)
+            .filter((m) => filteredLocalMeasures.includes(measureLocalId(m)))
+            .map((measure) => {
+                const measureNumerator = modifySimpleMeasure(measure, (m) => m.noRatio().defaultLocalId());
+                return [measureLocalId(measure), measureNumerator];
+            }),
+    );
+}
+
+/**
+ * Converts internal `afmFilters` to backend AFM filters. Those filters that reference some compute ratio measure from
+ * `afmMeasures` will be handled specially (see `determineComputeRatioMeasureNumerators`) and will also lead to `auxMeasures`
+ * being generated (intended to be used by the callers in crafting the final backend AFM).
+ */
+export function convertAfmFilters(
+    afmMeasures: IMeasure[],
+    afmFilters: IFilter[],
+): { filters: FilterDefinition[]; auxMeasures: MeasureItem[] } {
+    const computeRatioMeasureNumerators = determineComputeRatioMeasureNumerators(afmMeasures, afmFilters);
+    const transformedFilters = afmFilters.map((filter) => {
+        if (isMeasureValueFilter(filter)) {
+            const [filteredMeasure, applyOnResult] = getActualFilteredMeasureInfo(
+                computeRatioMeasureNumerators,
+                filter.measureValueFilter.measure,
+            );
+            const { condition } = filter.measureValueFilter;
+            const transformedFilter = { measureValueFilter: { measure: filteredMeasure, condition } };
+            return newFilterWithApplyOnResult(transformedFilter, applyOnResult);
+        } else if (isRankingFilter(filter)) {
+            const [filteredMeasure, applyOnResult] = getActualFilteredMeasureInfo(
+                computeRatioMeasureNumerators,
+                filter.rankingFilter.measure,
+            );
+            const { attributes, operator, value } = filter.rankingFilter;
+            const transformedFilter = {
+                rankingFilter: { measure: filteredMeasure, attributes, operator, value },
```

Comment:
small stuff: you should be able to use the `newRankingFilter` factory function to create this transformed IRankingFilter

Reply:
It's not that easy because `attributes` can be undefined which is not something the factory method counts with. I now see that I could make it with `attributes || []` because in `newRankingFilter` there is
```
        const attributesProp = attributeRefs.length ? { attributes: attributeRefs } : {};
```
but it seeems to me little bit clumsy. Besides, it's IMO wrong to treat empty attributes the same as no attributes...

Reply:
i see now that the factory is a bit unfortunate. keep it as is for now, just add a TODO telling that this needs to be addressed by first-class support in sdk-model.